File: 69/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        while l <= r:
            mid = (l + r) // 2
            val = mid ** 2
            if val == x:
                return mid
            elif val < x:
                l = mid + 1
            else:
                r = mid - 1
        return r

File: 69/Solution.java
class Solution {
    public int mySqrt(int x) {
        int l = 0, r = x;
        while (l <= r) {
            int mid = (l + r) / 2;
            long val = (long) mid * mid;
            if (val == x) {
                return mid;
            } else if (val < x) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return r;
    }
}